### DISCUSSION TITLE
I need Qt6::WebengineCore, but that requires webengine_proprietary_codecs to be set to "ON", and it isn't in the Homebrew Qt6

### DISCUSSION THREAD

**Discussion by CvR42** [Homebrew/discussions#3671] [Everyday usage]
I'm trying to port a program to Qt6, and that program requires the `WebEngineWidgets` component. Unfortunately, it seems that Homebrew Qt6 is compiled with the wrong flags for this, see the cmake output below. Is there an easy workaround for this so that I can use the homebrew compiled library, or should I just download and compile the library myself?

```
/Applications/CLion.app/Contents/bin/cmake/mac/bin/cmake -DCMAKE_BUILD_TYPE=Debug -DCMAKE_MAKE_PROGRAM=/Applications/CLion.app/Contents/bin/ninja/mac/ninja -G Ninja -S /Users/keesreeuwijk/CLionProjects/QPinner -B /Users/keesreeuwijk/CLionProjects/QPinner/cmake-build-debug
CMake Error at /usr/local/lib/cmake/Qt6/QtFeature.cmake:1249 (message):
  Feature webengine_proprietary_codecs is already defined to be "OFF" and
  should now be set to "ON" when importing features from Qt6::WebEngineCore.
Call Stack (most recent call first):
  /usr/local/lib/cmake/Qt6WebEngineCore/Qt6WebEngineCoreConfig.cmake:118 (qt_make_features_available)
  /Applications/CLion.app/Contents/bin/cmake/mac/share/cmake-3.23/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
  /usr/local/lib/cmake/Qt6/QtPublicDependencyHelpers.cmake:14 (find_dependency)
  /usr/local/lib/cmake/Qt6WebEngineWidgets/Qt6WebEngineWidgetsDependencies.cmake:96 (_qt_internal_find_dependencies)
  /usr/local/lib/cmake/Qt6WebEngineWidgets/Qt6WebEngineWidgetsConfig.cmake:40 (include)
  /usr/local/lib/cmake/Qt6/Qt6Config.cmake:213 (find_package)
  CMakeLists.txt:67 (find_package)


CMake Warning at /usr/local/lib/cmake/Qt6/Qt6Config.cmake:213 (find_package):
  Found package configuration file:

    /usr/local/lib/cmake/Qt6WebEngineWidgets/Qt6WebEngineWidgetsConfig.cmake

  but it set Qt6WebEngineWidgets_FOUND to FALSE so package
  "Qt6WebEngineWidgets" is considered to be NOT FOUND.
Call Stack (most recent call first):
  CMakeLists.txt:67 (find_package)
```

**Comment by carlocab** [comment]
It sounds like you may need to compile this yourself.

Setting `webengine_proprietary_codecs` to `ON` sounds like it has licensing implications that would make it a copyright violation to distribute compiled binaries.

See the warning [here](https://doc.qt.io/qt-6/qtwebengine-features.html).

**Comment by Bo98** [comment]
We already set this to ON so not sure what's happening here: https://github.com/Homebrew/homebrew-core/blob/master/Formula/qt.rb#L180

**Reply by CvR42** [reply]
@Bo98: Thanks for the reaction, at least the licencing is apparently not a problem.

I have tried to debug the issue by adding some message() statements to /usr/local/Cellar/qt/6.3.1_4/lib/cmake/Qt6/QtFeature.cmake. As far as I can tell, there is nowhere where this feature is explicitly set to "OFF", (there are only a few set() statements where that could happen, and I added message() statements before all of them). Thus, I strongly suspect that this "-DQT_FEATURE_webengine_proprietary_codecs=ON" in 'qt.rb' is for some reason not seen by the Qt6 configuration (rather than overridden). Then again, I'm sure you would have noticed if that happened, so I have no idea what is going on.

For what it is worth, I also noticed that spelling the value as 'On' or 'Off' in 'qt.rb' (rather than 'ON'), does not change the error message, the offending value is still spelled as 'OFF'. Oh, and I have also uninstalled 'qt' entirely and reinstalled it. Didn't help.

I'm still stuck.

**Comment by CvR42** [comment]
I am happy to report that with the latest qt6 and the latest CLion the problem has gone away. As far as I am concerned this is a 'case closed'.

What perhaps fixed the problem is that (as suggested by CLion) I cleared the CMake cache. I'm pretty sure I cleared it before, but hey, whatever.
